uberfire-m2repo-editor/uberfire-m2repo-editor-backend/src/main/java/org/guvnor/m2repo/backend/server/FixNotMavenizedArtifactInfo.java

```diff
@@ -0,0 +1,61 @@
+package org.guvnor.m2repo.backend.server;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.Properties;
+import java.util.stream.Collectors;
+
+import org.guvnor.common.services.project.model.GAV;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.ARTIFACT_ID;
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.GROUP_ID;
+import static org.guvnor.m2repo.backend.server.M2ServletContextListener.VERSION;
+
+public class FixNotMavenizedArtifactInfo {
+
+    private static final Logger logger = LoggerFactory.getLogger(FixNotMavenizedArtifactInfo.class);
+
+    private final Properties notMavenizedArtifacts = new Properties();
+    private final String pomTemplate;
+
+    public FixNotMavenizedArtifactInfo() {
+        String pomTemplate = null;
+        try {
+            pomTemplate = new BufferedReader(new InputStreamReader(M2ServletContextListener.class.getResourceAsStream("template.pom")))
+                    .lines().collect(Collectors.joining("\n"));
+            final InputStream isNotMavenizedArtifacts = M2ServletContextListener.class.getResourceAsStream("/not-mavenized-artifacts.txt");
+            if (isNotMavenizedArtifacts != null) {
+                notMavenizedArtifacts.load(isNotMavenizedArtifacts);
+            }
+        } catch (IOException e) {
+            logger.error(e.getMessage(), e);
+        }
+        this.pomTemplate = pomTemplate;
+    }
+
+    public Properties getProperties(final String filePath) {
+        final Properties result = new Properties();
+        final String fullFileName = filePath.substring(filePath.lastIndexOf("/") + 1);
```

Comment:
Whenever you're doing `(last)IndexOf(String)` with just one character String, youshould replace it with overloaded version `(last)IndexOf(Char)`, which is faster. 3x in this PR